File: leetcode/2769-find-the-maximum-achievable-number/theMaximumAchievableX_test.go
package leetcode

import (
	"fmt"
	"path/filepath"
	"testing"

	"github.com/edram/leetcode/util"
)

type TheMaximumAchievableXCase struct {
	Input struct {
		Num int
		T   int
	}

	Output int
}

func TestTheMaximumAchievableX(t *testing.T) {
	var casePath, _ = filepath.Abs("./cases.json")
	payload, err := util.LoadCases[[]TheMaximumAchievableXCase](casePath)
	if err != nil {
		panic(err)
	}

	for i, testCase := range *payload {
		fmt.Printf("正在执行第 %d 个测试用例\n", i+1)
		fmt.Printf("string: %v ==> %v \n", testCase.Input, testCase.Output)

		result := theMaximumAchievableX(testCase.Input.Num, testCase.Input.T)
		if result != testCase.Output {
			t.Errorf("期望 %v, 结果 %v", testCase.Output, result)
		}
	}
}

File: leetcode/2769-find-the-maximum-achievable-number/theMaximumAchievableX.test.ts
import cases from "./cases.json";
import theMaximumAchievableX from "./theMaximumAchievableX";

test("find the maximum achievable number", () => {
  cases.forEach((testCase, index) => {
    console.log(`正在执行第 ${index + 1} 个测试用例\n`);
    const result = theMaximumAchievableX(
      testCase.input.num,
      testCase.input.t,
    );
    console.log(
      `input: ${testCase.input} , output: ${testCase.output} ==> ${result} \n`
    );

    expect(result).toEqual(testCase.output);
  });
});
